### ISSUE TITLE
Feature Request: Publish TypeScript Types

### ISSUE THREAD

**Issue by cinderblock** [Fridus/webpack-watch-files-plugin#14]
Webpack supports `webpack.config.ts` these days. It would be very helpful to publish some typings. Ideally, these would be bundled in this package in the published sources.

Alternatively, one could (independently) publish to DefinitelyTyped to provide the types.

What is the appetite here for either:
1. Converting `src` to TypeScript
2. Adding hand created `.d.ts` files

**Comment by bengry** [comment]
I've addressed this in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56102.

The library could also be migrated to TypeScript, or include the type definitions within the package, but this is a good first step at least, especially considering there haven't been any commits to this repo for the past ~1 year (although it's still working perfectly fine for me, even with Webpack 5).

**Comment by cinderblock** [comment]
@bengry Nice!

I, personally, strongly feel that types should be directly included in published repos. Why force developers to download/add a second dependency? (This is rhetorical)

I'm holding out for this package to incorporate the types fully. 🙏

**Comment by bengry** [comment]
> @bengry Nice!
> 
> I, personally, strongly feel that types should be directly included in published repos. Why force developers to download/add a second dependency? (This is rhetorical)
> 
> I'm holding out for this package to incorporate the types fully. 🙏

@cinderblock I'd love to be wrong here.
@Fridus Please let me know if you'd like me to make a PR with the types to this repo - they're already merged over at DefinitelyTyped (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56102), but that can easily be changed to be a direct proxy to the types published from this package (same as what `@types/moment` and `@types/webpack@^5` are doing).

**Comment by cinderblock** [comment]
@bengry You are not wrong to publish to DT, especially considering the author of this package is not speaking up.
